### DISCUSSION TITLE
Insert all text into new page with origin properties

### DISCUSSION THREAD

**Discussion by binodil** [pymupdf/PyMuPDF#2326] [Looking for help] [answered]
Hi there!

I have extracted the word from the origin page with `get_text("words")`.  The issue is that `get_text('words')` returns a list of [x0,y0,x1,y1,text, line and etc.]

When I want to add that words with `Shape.insert_text()` the font size and other properties become the default.

## What technique do you suggest to exactly insert the text from one page to another page with their own properties?

I hope my explanation was short and clear!

Looking forward to your reply!

**Comment by JorjMcKie** [comment, marked as answer]
The best start is going from `page.get_text("dict")`. This is a dictionary of stacked dictionaries described in detail [here](https://pymupdf.readthedocs.io/en/latest/textpage.html#structure-of-dictionary-outputs). It contains a very complete information of text properties, including text color.

If you want to see a sophisticated example, look at the font replacing scripts [here](https://github.com/pymupdf/PyMuPDF-Utilities/tree/master/font-replacement).

The "words" extraction is of limited use here, because you do not have the font and the insertion point.
